ccf_transaction_processor/scripts/register_enclave_attestation_verification_policy.py

```diff
@@ -0,0 +1,128 @@
+def Main() :
+    default_config = os.path.join(CCF_Etc, 'cchost.toml')
+    default_output = os.path.join(CCF_Keys, 'ledger_authority.pem')
+
+    parser = argparse.ArgumentParser(description='Fetch the ledger authority key from a CCF server')
+
+    parser.add_argument('--logfile', help='Name of the log file, __screen__ for standard output', default='__screen__', type=str)
+    parser.add_argument('--loglevel', help='Logging level', default='INFO', type=str)
+
+    parser.add_argument('--ccf-config', help='Name of the CCF configuration file', default = default_config, type=str)
+    parser.add_argument('--member-name', help="Name of the user being added", default = "memberccf", type=str)
+
+    parser.add_argument('--check_attestation', default = False, help="enable attestation verification", action='store_true')
+    parser.add_argument('--mrenclave', help="Expected MRENCLAVE of pdo enclaves", type=str)
+    parser.add_argument('--basename', help="Pdo enclave basename", type=str)
+    parser.add_argument('--ias-public-key', help="IAS public derived from IAS cert used to verify singature on IAS reports", type=str)
+
+    options = parser.parse_args()
+
+    # -----------------------------------------------------------------
+    LOG.remove()
+    if options.logfile == '__screen__' :
+        LOG.add(sys.stderr, level=options.loglevel)
+    else :
+        LOG.add(options.logfile)
+
+    # -----------------------------------------------------------------
+    try :
+        config = toml.load(options.ccf_config) #If config exists, it is used to set ledger host, port.
+                                               #else we read from env PDO_LEDGER_URL
+                                               #both cases are used. for end-to-end docker testing, we rely on config
+                                               #for tp in docker, and pdo in bare-metal, we do not rely on config
+                                               #(since tp config is not created unless during tp installation)
+    except :
+        config = None
+
+    host, port = parse_ledger_url(config)
+    LOG.info(host)
+
+    network_cert = os.path.join(CCF_Keys, "networkcert.pem")
+    member_cert = os.path.join(CCF_Keys, "{}_cert.pem".format(options.member_name))
+    member_key = os.path.join(CCF_Keys, "{}_privk.pem".format(options.member_name))
+
+    try:
+        member_client = CCFClient(
+            host,
+            port,
+            network_cert,
+            session_auth=Identity(member_key, member_cert, "member"),
+            signing_auth=Identity(member_key, member_cert, "member"),
+        )
+    except Exception as e:
+        LOG.error('failed to connect to CCF service : {}'.format(str(e)))
+        sys.exit(-1)
+
+    if options.check_attestation:
+        if (options.mrenclave is None) or (options.basename is None) or (options.ias_public_key is None):
+            raise Exception("Please provide all of the exepected params to set attestation policy")
```

Comment:
Because this is a command-line argument error, you might want to consider changing this to a `parser.error()` call instead.

Reply:
the scripts folder is not currently packaged, but we might very well end up doing that in future, depending on "deployment templates".  Honestly, I would prefer the start_ccf_network.sh to call a single python script that can in turn invoke each of tthe separate ccf member commands. The reason why we get away for now is that these commands are mutually independent, (there is no need to use the output of one command as the input of another one.),